bblocks/import_tools/unaids.py

```diff
@@ -0,0 +1,242 @@
+            df = self.__extract_data(indicator)
+            self.indicators[indicator] = df
+            df.to_csv(
+                f"{PATHS.imported_data}/aids_{self.grouping}_{indicator}.csv",
+                index=False,
+            )
+
+        self.indicators[indicator] = pd.read_csv(
+            f"{PATHS.imported_data}/aids_{self.grouping}_{indicator}.csv"
+        )
+
+        return self
+
+    def __extract_data(self, indicator) -> pd.DataFrame:
+        """Extract and clean aids data"""
+
+        category = get_category(self.available_indicators(), indicator)
+        response = get_response(
+            indicator, category, self.__group_name, self.__group_code
+        )
+
+        dimensions = get_dimensions(response)
+        years = get_years(response)
+
+        df = parse_data_table(response, dimensions, years)
+        if self.grouping == "region":
+            df = pd.concat([df, parse_global_data(response, dimensions, years)])
+
+        return clean_data(df, indicator)
+
+    def update(self, reload_data: bool = True):
+        """Update all loaded indicators saved on the disk
+
+        When called, it will go through each loaded indicator
+        and update the data saved on disk.
+
+        Args:
+            reload_data (bool): If True, the updated data will be reloaded
+             into the object.
+
+        Returns:
+            The same object to allow chaining
+        """
+
+        if len(self.indicators) == 0:
+            raise RuntimeError("No indicators loaded")
+
+        for indicator in self.indicators:
+            df = self.__extract_data(indicator)
+            df.to_csv(
+                f"{PATHS.imported_data}/aids_{self.grouping}_{indicator}", index=False
+            )
+
+            if reload_data is True:
+                self.indicators[indicator] = df
+
+        return self
+
+    def get_data(self, indicators: Optional[str | list] = None) -> pd.DataFrame:
+        """Get the data as a Pandas DataFrame
+
+        Args:
+            indicators:  By default, all indicators are returned in a single DataFrame.
+            If a list of indicators is passed, only those indicators will be returned.
+            A single indicator can be passed as a string as well.
+
+        Returns:
+            A Pandas DataFrame with the requested indicator data
+        """
+
+        if len(self.indicators) == 0:
+            raise RuntimeError("No indicators loaded")
+
+        if indicators is None:
+            return pd.concat(self.indicators.values())
+
+        if isinstance(indicators, list):
+            df = pd.concat([self.indicators[i] for i in indicators])
+            for i in indicators:
+                if i not in self.available_indicators().indicator:
+                    warnings.warn(f"Invalid indicator: {i}")
+
+            return df
+
+        if indicators not in self.indicators:
+            raise ValueError(f"Indicator not loaded: {indicators}")
```

Comment:
isn't this check effectively what the loop just above does? If so, that should be a check you perform right after you check that indicators is not None